core/chains/evm/txm/storage/inmemory_store_test.go

```diff
@@ -0,0 +1,482 @@
+
+	assert.Equal(t, 0, m.CountUnstartedTransactions())
+
+	insertUnstartedTransaction(m)
+	assert.Equal(t, 1, m.CountUnstartedTransactions())
+}
+
+func TestCreateEmptyUnconfirmedTransaction(t *testing.T) {
+	t.Parallel()
+
+	fromAddress := testutils.NewAddress()
+	m := NewInMemoryStore(logger.Test(t), fromAddress, testutils.FixtureChainID)
+	_, err := insertUnconfirmedTransaction(m, 0)
+	require.NoError(t, err)
+
+	t.Run("fails if unconfirmed transaction with the same nonce exists", func(t *testing.T) {
+		_, err := m.CreateEmptyUnconfirmedTransaction(0, 0)
+		require.Error(t, err)
+	})
+
+	t.Run("creates a new empty unconfirmed transaction", func(t *testing.T) {
+		tx, err := m.CreateEmptyUnconfirmedTransaction(1, 0)
+		require.NoError(t, err)
+		assert.Equal(t, types.TxUnconfirmed, tx.State)
+	})
+}
+
+func TestCreateTransaction(t *testing.T) {
+	t.Parallel()
+
+	fromAddress := testutils.NewAddress()
+
+	t.Run("creates new transactions", func(t *testing.T) {
+		m := NewInMemoryStore(logger.Test(t), fromAddress, testutils.FixtureChainID)
+		now := time.Now()
+		txR1 := &types.TxRequest{}
+		txR2 := &types.TxRequest{}
+		tx1 := m.CreateTransaction(txR1)
+		assert.Equal(t, uint64(1), tx1.ID)
+		assert.LessOrEqual(t, now, tx1.CreatedAt)
+
+		tx2 := m.CreateTransaction(txR2)
+		assert.Equal(t, uint64(2), tx2.ID)
+		assert.LessOrEqual(t, now, tx2.CreatedAt)
+
+		assert.Equal(t, 2, m.CountUnstartedTransactions())
+	})
+
+	t.Run("prunes oldest unstarted transactions if limit is reached", func(t *testing.T) {
+		m := NewInMemoryStore(logger.Test(t), fromAddress, testutils.FixtureChainID)
+		overshot := 5
+		for i := 1; i < maxQueuedTransactions+overshot; i++ {
+			r := &types.TxRequest{}
+			tx := m.CreateTransaction(r)
+			//nolint:gosec // this won't overflow
+			assert.Equal(t, uint64(i), tx.ID)
+		}
+		// total shouldn't exceed maxQueuedTransactions
+		assert.Equal(t, maxQueuedTransactions, m.CountUnstartedTransactions())
+		// earliest tx ID should be the same amount of the number of transactions that we dropped
+		tx, err := m.UpdateUnstartedTransactionWithNonce(0)
+		require.NoError(t, err)
+		//nolint:gosec // this won't overflow
+		assert.Equal(t, uint64(overshot), tx.ID)
+	})
+}
+
+func TestFetchUnconfirmedTransactionAtNonceWithCount(t *testing.T) {
+	t.Parallel()
+
+	fromAddress := testutils.NewAddress()
+	m := NewInMemoryStore(logger.Test(t), fromAddress, testutils.FixtureChainID)
+
+	tx, count := m.FetchUnconfirmedTransactionAtNonceWithCount(0)
+	assert.Nil(t, tx)
+	assert.Equal(t, 0, count)
+
+	var nonce uint64
+	_, err := insertUnconfirmedTransaction(m, nonce)
+	require.NoError(t, err)
+	tx, count = m.FetchUnconfirmedTransactionAtNonceWithCount(0)
+	assert.Equal(t, *tx.Nonce, nonce)
+	assert.Equal(t, 1, count)
+}
+
+func TestMarkTransactionsConfirmed(t *testing.T) {
+	t.Parallel()
+
+	fromAddress := testutils.NewAddress()
+
+	t.Run("returns 0 if there are no transactions", func(t *testing.T) {
```

Comment:
Could you also add a no-op test case with transactions in storage? Something like
- Nonce: 0, Confirmed
- Nonce: 1, Unconfirmed
- Latest nonce: 1